lib/util/template/index.ts

```diff
@@ -155,20 +156,13 @@ const prBodyFields = [
 
 const handlebarsUtilityFields = ['else'];
 
-const allowedFieldsList = Object.keys(allowedFields)
-  .concat(exposedConfigOptions)
-  .concat(prBodyFields)
-  .concat(handlebarsUtilityFields);
+const envObjectField = ['env'];
 
 type CompileInput = Record<string, unknown>;
 
-const allowedTemplateFields = new Set([
-  ...Object.keys(allowedFields),
-  ...exposedConfigOptions,
-]);
-
 const compileInputProxyHandler: ProxyHandler<CompileInput> = {
   get(target: CompileInput, prop: keyof CompileInput): unknown {
+    const allowedTemplateFields = getAllowedTemplateFields();
```

Comment:
this is very bad for perf. we should refactor the code to pass from initial compile function